src/Components/Web.JS/src/Services/ComponentDescriptorDiscovery.ts

```diff
@@ -26,6 +26,34 @@ function discoverServerComponents(document: Document): ServerComponentDescriptor
   return discoveredComponents.sort((a, b): number => a.sequence - b.sequence);
 }
 
+const blazorStateCommentRegularExpression = /\W*Blazor-Component-State:(?<state>[a-zA-Z0-9\+\/=]+)$/;
+
+export function discoverPersistedState(node: Node): string {
+  if (node.nodeType === Node.COMMENT_NODE) {
+    const content = node.textContent || '';
+    const stateCommentRegex = new RegExp(blazorStateCommentRegularExpression);
+    const parsedState = stateCommentRegex.exec(content);
```

Comment:
I know it's a bit early to do nitpick comments, but just don't want to forget this :)

```suggestion
    const parsedState = blazorStateCommentRegularExpression.exec(content);
```

It's already a `RegExp` I think.

Reply:
You are right, I think I copied this from the other comment parsing code, so they must be both wrong.